akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -31,6 +32,31 @@ object Sink {
     new Sink(scaladsl.Sink.fold[U, In](zero)(f.apply).toCompletionStage())
 
   /**
+   * A `Sink` that will use Java 8 `Collector` to reduce and optionally transform
+   * accumulated result into a final representation.
+   * As collector is mutable you can use it only with one stream.
+   * Reduction operations performs sequentially.
+   */
+  def javaCollector[T, A, R](collector: java.util.stream.Collector[T, A, R]): Sink[T, CompletionStage[R]] =
+    new Sink(scaladsl.Sink.javaCollector[T, A, R](collector).toCompletionStage())
```

Comment:
At that point, where the original PR was opened on which this is based (maybe in January?), there was a discussion, but I don't recall what we ended up with :) I am fine with whatever the majority likes.